cowprotocol/cowswap
# feat(limit): max limit order deadline

# Summary

Part of 2024 roadmap, one low hanging fruit is to allow users to easily set `max` limit order duration on creation.

Also, the current max deadline we had didn't quite match what's set on the backend, leading to error when placing the order.

This change updates the value to match [backend's value](https://github.com/cowprotocol/infrastructure/blob/901ed8e2fe3ea57956585f107bdd7539c2e7d3d1/services/Pulumi.yaml#L15) and adds the `max` option in the date picker.

![image](https://github.com/cowprotocol/cowswap/assets/43217/210ecb78-de8c-4362-b108-4b671692c136)

# To Test

1. Open limit orders tab
2. Open the expiry dropdown
* `Max` should be available
3. Pick `Max`
4. Place order
* Order should be successfully placed
* Duration should be 6 months from time of placement
